### ISSUE TITLE
Integration fails after upgrading

### ISSUE THREAD

**Issue by 0utc45t** [mawinkler/astroweather#64]
After upgrading both HA and Astroweather, Astroweather no long works. I tried deleting and reinstalling integration, it did not help. Debug log snippet under.

2024-11-18 19:51:42.602 WARNING (MainThread) [pyastroweatherio.client] Column 'dew_point_temperature' has NaN or None: True. Calculating dew points.
2024-11-18 19:51:42.812 ERROR (MainThread) [custom_components.astroweather] Unexpected error fetching astroweather data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 279, in _async_update_data
    return await self.update_method()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 165, in get_location_data
    return await self._get_location_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 545, in _get_location_data
    "moon_data": await self._astro_routines.moon_data(),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/helper_functions.py", line 1030, in moon_data
    return MoonData(data=md)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/dataclasses.py", line 123, in __init__
    def __init__(self, *, data: MoonDataModel):
  File "/usr/local/lib/python3.12/site-packages/typeguard/_functions.py", line 137, in check_argument_types
    check_type_internal(value, annotation, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal
    checker(value, origin_type, args, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 276, in check_typed_dict
    raise TypeCheckError(f"is missing required key(s): {keys_formatted}")
typeguard.TypeCheckError: argument "data" (dict) is missing required key(s): "next_rising", "next_setting", "previous_rising", "previous_setting"

**Comment by mawinkler** [comment]
Looks like you're in a place where the moon isn't currently rising or setting? If so, I seem to be missing a control for that. Can you give me an approximate location that I can use to check this?

**Comment by 0utc45t** [comment]
OK,
65°00'47.9"N 25°27'54.8"E

**Comment by mawinkler** [comment]
Now I remember this location ;-)
I identified the problem and will find a way to properly solve this.

**Comment by mawinkler** [comment]
Could you please change to `pyastroweatherio` to version `0.70.2` in your manifest.json?

```json
{
...
  "requirements": ["pyastroweatherio==0.70.2"],
...
}
```

**Comment by 0utc45t** [comment]
Reloading integration was not enough, but after full reboot of HA the integration seems to work again. Thanks!

**Comment by mawinkler** [comment]
🌈 🦄 
Great. Your location is now in the list of test-locations. Will do a new release with the fixed library tonight.

**Comment by renurios** [comment]
I got an similar problem since upgrading.

Dieser Fehler stammt von einer benutzerdefinierten Integration

Logger: custom_components.astroweather
Quelle: helpers/update_coordinator.py:382
Integration: AstroWeather (Dokumentation, Probleme)
Erstmals aufgetreten: 15:20:05 (11 Vorkommnisse)
Zuletzt protokolliert: 15:27:44

Unexpected error fetching astroweather data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 279, in _async_update_data
    return await self.update_method()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 165, in get_location_data
    return await self._get_location_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 551, in _get_location_data
    "uptonight": await self._get_deepsky_objects(),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 845, in _get_deepsky_objects
    items.append(UpTonightDSOData(data=item))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/dataclasses.py", line 386, in __init__
    def __init__(self, *, data: UpTonightDSODataModel):
  File "/usr/local/lib/python3.12/site-packages/typeguard/_functions.py", line 137, in check_argument_types
    check_type_internal(value, annotation, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal
    checker(value, origin_type, args, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 282, in check_typed_dict
    check_type_internal(argvalue, argtype, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal
    checker(value, origin_type, args, memo)
  File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 631, in check_number
    raise TypeCheckError("is neither float or int")
typeguard.TypeCheckError: value of key 'visual_magnitude' of argument "data" (dict) is neither float or int

**Comment by 0utc45t** [comment]
Offtopic request: Would it be possible to have uptonight installable from HACS?  I've looked at it a couple of times here at github... but I don't quite know how to install it in my Yellow running HA. I'm afraid I'll mess up my HA installation by trying to do it manually.

**Comment by mawinkler** [comment]
> I got an similar problem since upgrading.
> 
> Dieser Fehler stammt von einer benutzerdefinierten Integration
> 
> Logger: custom_components.astroweather Quelle: helpers/update_coordinator.py:382 Integration: AstroWeather (Dokumentation, Probleme) Erstmals aufgetreten: 15:20:05 (11 Vorkommnisse) Zuletzt protokolliert: 15:27:44
> 
> Unexpected error fetching astroweather data Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh self.data = await self._async_update_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 279, in _async_update_data return await self.update_method() ^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 165, in get_location_data return await self._get_location_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 551, in _get_location_data "uptonight": await self._get_deepsky_objects(), ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 845, in _get_deepsky_objects items.append(UpTonightDSOData(data=item)) ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/dataclasses.py", line 386, in **init** def **init**(self, *, data: UpTonightDSODataModel): File "/usr/local/lib/python3.12/site-packages/typeguard/_functions.py", line 137, in check_argument_types check_type_internal(value, annotation, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 282, in check_typed_dict check_type_internal(argvalue, argtype, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 631, in check_number raise TypeCheckError("is neither float or int") typeguard.TypeCheckError: value of key 'visual_magnitude' of argument "data" (dict) is neither float or int

So this is related to Uptonight. Can it be that you're using a pretty old version of it? The magnitude information of deep sky objects seem to be missing in Uptonights output.
Nevertheless, I need to be a little more fault-tolerant on the Astroweather side...

**Comment by mawinkler** [comment]
> I got an similar problem since upgrading.
> 
> Dieser Fehler stammt von einer benutzerdefinierten Integration
> 
> Logger: custom_components.astroweather Quelle: helpers/update_coordinator.py:382 Integration: AstroWeather (Dokumentation, Probleme) Erstmals aufgetreten: 15:20:05 (11 Vorkommnisse) Zuletzt protokolliert: 15:27:44
> 
> Unexpected error fetching astroweather data Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh self.data = await self._async_update_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 279, in _async_update_data return await self.update_method() ^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 165, in get_location_data return await self._get_location_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 551, in _get_location_data "uptonight": await self._get_deepsky_objects(), ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 845, in _get_deepsky_objects items.append(UpTonightDSOData(data=item)) ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/dataclasses.py", line 386, in **init** def **init**(self, *, data: UpTonightDSODataModel): File "/usr/local/lib/python3.12/site-packages/typeguard/_functions.py", line 137, in check_argument_types check_type_internal(value, annotation, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 282, in check_typed_dict check_type_internal(argvalue, argtype, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 631, in check_number raise TypeCheckError("is neither float or int") typeguard.TypeCheckError: value of key 'visual_magnitude' of argument "data" (dict) is neither float or int

Could you please change `pyastroweatherio` to version `0.70.3` in your manifest.json?

```json
{
...
  "requirements": ["pyastroweatherio==0.70.3"],
...
}
```

**Comment by renurios** [comment]
> > I got an similar problem since upgrading.
> > Dieser Fehler stammt von einer benutzerdefinierten Integration
> > Logger: custom_components.astroweather Quelle: helpers/update_coordinator.py:382 Integration: AstroWeather (Dokumentation, Probleme) Erstmals aufgetreten: 15:20:05 (11 Vorkommnisse) Zuletzt protokolliert: 15:27:44
> > Unexpected error fetching astroweather data Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh self.data = await self._async_update_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 279, in _async_update_data return await self.update_method() ^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 165, in get_location_data return await self._get_location_data() ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 551, in _get_location_data "uptonight": await self._get_deepsky_objects(), ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/client.py", line 845, in _get_deepsky_objects items.append(UpTonightDSOData(data=item)) ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/pyastroweatherio/dataclasses.py", line 386, in **init** def **init**(self, *, data: UpTonightDSODataModel): File "/usr/local/lib/python3.12/site-packages/typeguard/_functions.py", line 137, in check_argument_types check_type_internal(value, annotation, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 282, in check_typed_dict check_type_internal(argvalue, argtype, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 946, in check_type_internal checker(value, origin_type, args, memo) File "/usr/local/lib/python3.12/site-packages/typeguard/_checkers.py", line 631, in check_number raise TypeCheckError("is neither float or int") typeguard.TypeCheckError: value of key 'visual_magnitude' of argument "data" (dict) is neither float or int
> 
> So this is related to Uptonight. Can it be that you're using a pretty old version of it? The magnitude information of deep sky objects seem to be missing in Uptonights output. Nevertheless, I need to be a little more fault-tolerant on the Astroweather side...

I got uptonight updated now its working again, sorry I didn't realise that it was related to uptonight😅

**Comment by 0utc45t** [comment]
> > Offtopic request: Would it be possible to have uptonight installable from HACS? I've looked at it a couple of times here at github... but I don't quite know how to install it in my Yellow running HA. I'm afraid I'll mess up my HA installation by trying to do it manually.
> 
> Actually not. Uptonight uses AstroPy, which cannot run asynchronously.

Oh, bummer. Then maybe little more detailed instructions on installing uptonight via commandline? I tried to run the pip install -r requirements.txt command and also the docker build -t uptonight . command, but both failed. I suspect I should have downloaded the repository first, but then again, I'm not sure to which directory. I think the answer is pretty simple, if one has more experience with tinkering HA with CLI, but then again, it's also very easy to mess the system up with CLI if you don't know what you're doing... Well, in any case, thanks for your work with astroweather. :-)

**Comment by mawinkler** [comment]
> > > Offtopic request: Would it be possible to have uptonight installable from HACS? I've looked at it a couple of times here at github... but I don't quite know how to install it in my Yellow running HA. I'm afraid I'll mess up my HA installation by trying to do it manually.
> > 
> > 
> > Actually not. Uptonight uses AstroPy, which cannot run asynchronously.
> 
> Oh, bummer. Then maybe little more detailed instructions on installing uptonight via commandline? I tried to run the pip install -r requirements.txt command and also the docker build -t uptonight . command, but both failed. I suspect I should have downloaded the repository first, but then again, I'm not sure to which directory. I think the answer is pretty simple, if one has more experience with tinkering HA with CLI, but then again, it's also very easy to mess the system up with CLI if you don't know what you're doing... Well, in any case, thanks for your work with astroweather. :-)

The easiest would be to run

```
git clone https://github.com/mawinkler/uptonight.git
cd uptonight
docker build -t uptonight .
```

assuming you have docker.

**Comment by mawinkler** [comment]
Fixed by release [v0.70.2](https://github.com/mawinkler/astroweather/releases/tag/v0.70.2).
